packages/modelviewer.dev/data/docs.json

```diff
@@ -1122,9 +1122,9 @@
         ]
       },
       {
-        "name": "createVideoTexture(uri)",
+        "name": "createVideoTexture(uri, options?)",
         "htmlName": "createVideoTexture",
-        "description": "Returns a Texture object for use with the <code>setTexture</code> method on <code>TextureInfo</code> objects in the Materials API. The video element created is available through <code>texture.source.element</code> for controlling playback.",
+        "description": "Returns a Texture object for use with the <code>setTexture</code> method on <code>TextureInfo</code> objects in the Materials API. The video element created is available through <code>texture.source.element</code> for controlling playback. for cross origin resources, you can set `options.crossOrigin` to `anonymous`.",
```

Comment:
Looking at this more closely, I wonder if we need a new parameter at all. We already have an attribute `withCredentials` that is used for all of our image loaders. In three.js it's doing: `credentials: this.withCredentials ? 'include' : 'same-origin'`. Could we just do `video.crossOrigin = this.withCredentials ? 'with-credentials' : 'anonymous'` to be consistent?

Reply:
https://modelviewer.dev/docs/index.html#entrydocs-loading-attributes-withCredentials

Reply:
This is what I understand from your comment:
```html
<model-viewer with-credentials src="https://modelviewer.dev/shared-assets/models/sphere.glb" ... >
```
```js
videoTexture = modelViewerAnimated.createVideoTexture("https://cdn.glitch.global/68b42b4e-5a36-47f5-ab00-09026a7a73d0/lottie-logo.mp4?v=1705067796906");
// video.crossOrigin would be 'anonymous' as you said
```

Testing it locally gave me this error:
```
Access to fetch at 'https://modelviewer.dev/shared-assets/models/sphere.glb' from origin 'http://localhost:8002' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'.
```

Which addresses the problem in a scenario that the glb file is on the same origin but the video is not.

Reply:
@elalish For improving the consistency in the code, we can also change:
```
interface CreateVideoTextureOptions {
  withCredentials?: boolean
}
```

Reply:
Hmm, how common do you think it is that different assets are coming from different origins for MV users? I do see your point about generality, but the fact is that we've already got this single `with-credentials` switch that affects `src`, `createTexture`, and `lottieLoader`, without any complaints yet. I think I'd rather add video to that pattern for now, or else do a complete refactor if it proves to be a user pain point. What do you think?

Reply:
> how common do you think it is that different assets are coming from different origins for MV users? 

I'm not really sure as I'm a newbie in this library, but I believe most users would serve their assets alongside their application, and it would be okay for them. However, it may be concerning for users who have an application on origin A and are using a CMS, which may be on origin B, to serve an image, a video, or a Lottie file.

>  I think I'd rather add video to that pattern for now, or else do a complete refactor if it proves to be a user pain point.

I think that each resource having its option to be with credentials or not would be a more bulletproof solution. You're right; it may not be a user pain point now, and for the sake of consistency, we can do as you suggested, which covers the case for users that have all assets on a cross-origin source.

I will update the MR ASAP.